unit_tests/test_install_script.sh

```diff
@@ -0,0 +1,193 @@
+#!/usr/bin/env bash
+
+dir_path=$(cd "$(dirname "$0")" && pwd)
+# shellcheck source=/dev/null
+source "${dir_path}/extracted_functions.sh"
+yaml_config="$(dirname "$dir_path")/.yamllint.yaml"
+config_file="/etc/datadog-agent/datadog.yaml"
+security_agent_config_file="/etc/datadog-agent/security-agent.yaml"
+system_probe_config_file="/etc/datadog-agent/system-probe.yaml"
+
+### ensure_config_file_exists
+testEnsureExists() {
+  ensure_config_file_exists "sudo" "/etc/hosts" "root"
+  assertEquals 1 $?
+}
+
+testEnsureExistsWrongSudo() {
+  sudo rm /etc/datadog-agent/datadog.yaml
+  ensure_config_file_exists "sumo" $config_file "dd-agent"
+  assertEquals 125 $?
+}
+
+testEnsureExistsFailsWrongUser() {
+  sudo rm /etc/datadog-agent/datadog.yaml
+  ensure_config_file_exists "sudo" $config_file "datad0g-agent"
+  assertEquals 1 $?
+}
+
+testEnsureNotExists() {
+  sudo rm /etc/datadog-agent/datadog.yaml
+  ensure_config_file_exists "sudo" $config_file "dd-agent"
+  assertEquals 0 $?
+}
+
+### update_api_key
+testUpdateKey() {
+  sudo cp ${config_file}.example $config_file
+  update_api_key "sudo" "123" $config_file
+  yamllint -c "$yaml_config" --no-warnings $config_file
+  assertEquals 0 $?
+  sudo grep -w "^api_key: 123$" $config_file | sudo tee tmp > /dev/null
```

Comment:
It should be possible to write to /tmp/tee and dropping the need for `sudo` here (and while counting the number of lines)

Reply:
Can I do this in a follow-up PR ?